python_modules/dagster/dagster/_core/definitions/decorators/repository_decorator.py

```diff
@@ -118,7 +119,13 @@ def __call__(
         if isinstance(repository_definitions, list):
             bad_defns = []
             repository_defns = []
-            defer_repository_data = False
+            # If any CodeLocationStateMetadataValue is present, then it was set during code location
+            # construction and needs to be injected on reconstruction via RepositoryLoadData, so
+            # defer.
+            defer_repository_data = any(
+                isinstance(value, CodeLocationStateMetadataValue)
+                for value in self.metadata.values()
+            )
```

Comment:
What is the line of sight to eliminating this? I am fine with this as a temporary but long term we should be able to just have this code path return vanilla `RepositoryDefinition`.

Reply:
Just need to study the code more and find a convenient spot to construct `RepositoryLoadData` that doesn't rely on `PendingRepositoryDefinition`. Should be able to do it next week.

Reply:
Ok. Can you create linear task for that and place a link to it in a comment.